querybuilder/rule.go

```diff
@@ -19,36 +21,52 @@ type Rule struct {
 }
 
 // Evaluate function checks whether the dataset matches with rule
-func (r *Rule) Evaluate(dataset map[string]interface{}) bool {
-	var wg sync.WaitGroup
+func (r *Rule) Evaluate(dataset map[string]interface{}) (bool, error) {
 	var input, value interface{}
+	var wg sync.WaitGroup
+	var errg errgroup.Group
 
 	opr, ok := operator.GetOperator(r.Operator)
 	if !ok {
-		return false
+		return false, errors.Errorf("Invalid Operator %s", r.Operator)
```

Comment:
```suggestion
		return false, errors.Errorf("invalid Operator %s", r.Operator)
```

Reply:
since we treat errors as values, they will be used by the caller to add context, etc. so fo the caller, say he wants to prefix it with some context, the error string can't start with capital letters